runtime/src/launch/slurm-srun/launch-slurm-srun.c

```diff
@@ -324,7 +338,13 @@ static char* chpl_launch_create_command(int argc, char* argv[],
     // If we're buffering the output, set the temp output file name.
     // It's always <tmpDir>/binaryName.<jobID>.out.
     if (bufferStdout != NULL) {
-      sprintf(tmpStdoutFileNoFmt, "%s/%s.%s.out", tmpDir, argv[0], "$SLURM_JOB_ID");
+      char* format = "%s/%s.%s.out";
+      char* tempArg = "$SLURM_JOB_ID";
+      int tmpStdoutFileNoFmtLen = strlen(format) + strlen(tmpDir) + 
+                                    strlen(argv[0]) + strlen(tempArg);
+      tmpStdoutFileNoFmt = (char *)chpl_mem_allocMany(tmpStdoutFileNoFmtLen, 
+                                sizeof(char), CHPL_RT_MD_FILENAME, -1, 0);
+      sprintf(tmpStdoutFileNoFmt, format, tmpDir, argv[0], tempArg);
```

Comment:
In cases like this where you've computed and stored the buffer length, wouldn't it be safer / more future-proof to use `snprintf()` rather than `sprintf()` (to avoid buffer overrun in case someone changes one part of the malloc/snprintf pattern in the future and doesn't change the other part to match?).  Essentially, I'm curious whether there was a rationale for using `sprintf()` in some cases and `snprintf()` in others.

Reply:
@bradcray
> I'm curious whether there was a rationale for using sprintf() in some cases and snprintf() in others.

I kept the usage of ``snprintf()`` and ``sprintf()`` the same as in the code before changes (this PR). But, now since we have the lengths of the char arrays with us, it would be better (safer) to convert ``sprintf()`` to ``snprintf()``, if we have the lengths of the char array already stored. This would also be more robust to changes in future, rather than ``sprintf()``. I will make these changes.
Thanks,
R. Chinmay

Reply:
Your rationale makes sense, but if you're willing to make that change while here, I think that'd be great, thank you!